src/Language/PureScript/Parser/Declarations.hs

```diff
@@ -231,6 +237,7 @@ parseLocalDeclaration :: TokenParser Declaration
 parseLocalDeclaration = positioned (P.choice
                    [ parseTypeDeclaration
                    , parseValueDeclaration
+                   , parseBoundValueDeclaration
```

Comment:
Thanks for your explanation, now I understand. Will work on it.

1. Remove `parseBoundValueDeclaration`
2. Edit `parseValueDeclaration` like following:  
    ```haskell
    binders <- many1 parseBinderAtom
    case binders of
      VarBinder Ident : binders -> ... -- return ValueDeclaration
      [binder] -> ... -- return BoundValueDeclaration
      _ -> ... -- error
    ```

By the way, how should I throw error from parse functions? Thinking about `throwError`, but cannot find a case it's used with `TokenParser a`.